<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SandboxMainWindow</class>
 <widget class="QMainWindow" name="SandboxMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1176</width>
    <height>740</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="SceneViewWidget" name="primitivePreview">
       <property name="minimumSize">
        <size>
         <width>400</width>
         <height>400</height>
        </size>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Primitive ID:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="primitiveSelectorCombo">
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="groupCamera">
          <property name="title">
           <string>Camera</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Camera Position:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="cameraPositionX">
               <property name="minimum">
                <double>-50000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>50000.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="cameraPositionY">
               <property name="minimum">
                <double>-50000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>50000.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="cameraPositionZ">
               <property name="minimum">
                <double>-50000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>50000.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>Camera FOV:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="cameraFOV">
               <property name="minimum">
                <double>0.100000000000000</double>
               </property>
               <property name="maximum">
                <double>360.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Camera Yaw:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="cameraYaw"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Camera Pitch:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="cameraPitch"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>Camera Roll:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="cameraRoll"/>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupRender">
          <property name="title">
           <string>Rendering</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QCheckBox" name="primitiveDrawWireframeCheck">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Draw wireframe</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="primitiveDrawBoundingBox">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Draw bounding box</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1176</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen_ZIP"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen_ZIPO">
   <property name="text">
    <string>Open ZIP</string>
   </property>
  </action>
  <action name="actionOpen_ZIP">
   <property name="text">
    <string>Open ZIP</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SceneViewWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>SceneViewWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionOpen_ZIP</sender>
   <signal>triggered()</signal>
   <receiver>SandboxMainWindow</receiver>
   <slot>onSelectLevelToOpenRequested()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>587</x>
     <y>369</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>SandboxMainWindow</receiver>
   <slot>onExitRequested()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>587</x>
     <y>369</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onSelectLevelToOpenRequested()</slot>
  <slot>onExitRequested()</slot>
  <slot>onLevelLoaded()</slot>
  <slot>onLevelLoadFailed(const QString &amp;errorMessage)</slot>
 </slots>
</ui>
